Shared/Coordinator/MainCoordinator.swift

```diff
@@ -38,36 +38,27 @@ class MainCoordinator: Coordinator<Void> {
         self.launchAndDeepLinkTask?.cancel()
 
         self.launchAndDeepLinkTask = Task {
-            let launchStatus: LaunchStatus = await withCheckedContinuation { continuation in
+            let deepLink: DeepLinkable? = await withCheckedContinuation { continuation in
```

Comment:
Now the launch coordinator doesn't finish until it succeeds. So instead of a launch status, it finishes with an optional deep link.